packages/workbox-core/src/utils/pluginUtils.ts

```diff
@@ -0,0 +1,25 @@
+/*
+  Copyright 2018 Google LLC
+
+  Use of this source code is governed by an MIT-style
+  license that can be found in the LICENSE file or at
+  https://opensource.org/licenses/MIT.
+*/
+
+import '../_version';
+
+export interface Plugin {
```

Comment:
Could we add in some TypeScript definitions for the signatures of these various callbacks, and then say that each of those properties maps to the corresponding callback signature, instead of a generic `Function`?

(I'm happy to figure out the syntax for doing that if you're not sure, as I don't want to create more work for you.)

Reply:
I definitely think we can, and consider this initial conversion the first step in what will likely be more changes as I learn more about Typescript definition capabilities.

My expectation is as we convert more packages over, it'll start to become clear how we can update these plugin definitions to be more useful for consumers of them.